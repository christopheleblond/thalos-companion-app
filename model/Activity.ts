export interface Activity {
    id: string;
    name: string;
    filterable?: boolean;
    countable?: boolean;
    style: { color: string, backgroundColor: string };
}
