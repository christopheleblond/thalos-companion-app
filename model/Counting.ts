type Counts = { [key: string]: number }

export interface DayCounts {
    dayId: string;
    afternoon?: Counts,
    night?: Counts
}

