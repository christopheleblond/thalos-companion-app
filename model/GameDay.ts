
export interface GameDay {
    id: string,
    date: Date
}

