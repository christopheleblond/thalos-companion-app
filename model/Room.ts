
export interface Room {
    id: string;
    name: string;
    capacity?: number;
    virtual?: boolean;
}

export interface OpenCloseRoom {
    dayId: string;
    opener?: {
        id: string;
        name: string;
    };
    openAt: string;
    closer?: {
        id: string;
        name: string;
    };
}