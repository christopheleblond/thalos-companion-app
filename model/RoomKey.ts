export interface RoomKey {
    id: string,
    name: string,
    owner?: {
        id: string,
        name: string
    }
}