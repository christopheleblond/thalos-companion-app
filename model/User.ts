export interface User {
    id: string;
    firstName?: string;
    name?: string;
}