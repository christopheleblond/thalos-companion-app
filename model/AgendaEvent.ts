import { Activity } from "./Activity";
import { GameDay } from "./GameDay";
import { Room } from "./Room";
import { User } from "./User";

export interface AgendaEvent {
    id: string;
    title: string;
    dayId?: string;
    day: GameDay;
    start: string;
    durationInMinutes: number;
    startTime?: number;
    endTime?: number;
    roomId?: string;
    room?: Room;
    tables?: number;
    activityId?: string;
    activity?: Activity;
    description?: string;
    creator?: User;
}
