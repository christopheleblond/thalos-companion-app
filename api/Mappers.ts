import { ACTIVITIES } from "@/constants/Activities"
import { ROOMS } from "@/constants/Rooms"
import { AgendaEvent } from "@/model/AgendaEvent"
import { fromActivityId, fromGameDayId, fromRoomId } from "@/utils/Utils"

export const mapDtoToAgendaEvent = (id: string, json: any, rooms = ROOMS, activities = ACTIVITIES): AgendaEvent => {
    return {
        ...json,
        id,
        day: fromGameDayId(json['dayId']),
        room: fromRoomId(json['roomId'], rooms),
        activity: fromActivityId(json['activityId'], activities),
    }
}

export const mapAgendaEventToDto = (json: Partial<AgendaEvent>): Partial<AgendaEvent> => {
    return {
        dayId: json.dayId,
        activityId: json.activityId,
        roomId: json.roomId,
        creator: json.creator,
        description: json.description,
        durationInMinutes: json.durationInMinutes,
        start: json.start,
        title: json.title,
        tables: json.tables,
        startTime: json.startTime,
        endTime: json.endTime
    }
}