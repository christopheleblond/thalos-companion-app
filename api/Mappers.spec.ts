import { Activity } from "@/model/Activity"
import { Room } from "@/model/Room"
import { mapAgendaEventToDto, mapDtoToAgendaEvent } from "./Mappers"

describe('Mappers tests', () => {

    const rooms: Room[] = [
        {
            id: 'room1',
            name: 'Room A',
            capacity: 10
        },
        {
            id: 'room2',
            name: 'Room B',
            capacity: 15
        },
        {
            id: 'room3',
            name: 'Room C',
            capacity: 20
        },
    ]

    const activities: Activity[] = [
        {
            id: 'activity1',
            name: 'Activity A',
            style: {
                backgroundColor: 'yellow',
                color: 'white'
            }
        },
        {
            id: 'activity2',
            name: 'Activity B',
            style: {
                backgroundColor: 'red',
                color: 'white'
            }
        }
    ]

    it('mapDtoToAgendaEvent should map to event', () => {
        expect(mapDtoToAgendaEvent('id', {
            dayId: '2025-01-02',
            activityId: 'activity2',
            roomId: 'room1',
            creator: {
                id: 'user1'
            },
            description: 'description',
            durationInMinutes: 100,
            start: '15H',
            title: 'Test',
            tables: 4,
            startTime: 123456,
            endTime: 223456
        }, rooms, activities)).toMatchSnapshot();
    })

    it('mapAgendaEventToDto should map to dto', () => {
        expect(mapAgendaEventToDto({
            activityId: 'activity2',
            activity: {
                id: "activity2",
                name: "Activity B",
                style: {
                    backgroundColor: "red",
                    color: "white",
                },
            },
            creator: {
                id: "user1",
            },
            dayId: '2025-01-02',
            day: {
                date: new Date('2025-01-02T00:00:00.000Z'),
                id: "2025-01-02",
            },
            description: "description",
            durationInMinutes: 100,
            endTime: 223456,
            id: "id",
            roomId: 'room1',
            room: {
                capacity: 10,
                id: "room1",
                name: "Room A",
            },
            start: "15H",
            startTime: 123456,
            tables: 4,
            title: "Test",
        })).toMatchSnapshot()
    })
})