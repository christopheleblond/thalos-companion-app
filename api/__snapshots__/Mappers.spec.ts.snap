// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mappers tests mapAgendaEventToDto should map to dto 1`] = `
{
  "activityId": "activity2",
  "creator": {
    "id": "user1",
  },
  "dayId": "2025-01-02",
  "description": "description",
  "durationInMinutes": 100,
  "endTime": 223456,
  "roomId": "room1",
  "start": "15H",
  "startTime": 123456,
  "tables": 4,
  "title": "Test",
}
`;

exports[`Mappers tests mapDtoToAgendaEvent should map to event 1`] = `
{
  "activity": {
    "id": "activity2",
    "name": "Activity B",
    "style": {
      "backgroundColor": "red",
      "color": "white",
    },
  },
  "activityId": "activity2",
  "creator": {
    "id": "user1",
  },
  "day": {
    "date": 2025-01-02T00:00:00.000Z,
    "id": "2025-01-02",
  },
  "dayId": "2025-01-02",
  "description": "description",
  "durationInMinutes": 100,
  "endTime": 223456,
  "id": "id",
  "room": {
    "capacity": 10,
    "id": "room1",
    "name": "Room A",
  },
  "roomId": "room1",
  "start": "15H",
  "startTime": 123456,
  "tables": 4,
  "title": "Test",
}
`;
